models/issues/review.go

```diff
@@ -303,15 +309,38 @@ func CreateReview(ctx context.Context, opts CreateReviewOptions) (*Review, error
 		CommitID:     opts.CommitID,
 		Stale:        opts.Stale,
 	}
+
 	if opts.Reviewer != nil {
+		review.Type = opts.Type
 		review.ReviewerID = opts.Reviewer.ID
-	} else {
-		if review.Type != ReviewTypeRequest {
-			review.Type = ReviewTypeRequest
+
+		reviewCond := builder.Eq{"reviewer_id": opts.Reviewer.ID, "issue_id": opts.Issue.ID}
+		// make sure user review requests are cleared
+		if opts.Type != ReviewTypePending {
+			if _, err := sess.Where(reviewCond.And(builder.Eq{"type": ReviewTypeRequest})).Delete(new(Review)); err != nil {
+				return nil, err
+			}
 		}
+		// make sure if the created review gets dismissed no old review surface
+		if opts.Type == ReviewTypeApprove || opts.Type == ReviewTypeReject {
+			if _, err := sess.Where(reviewCond.And(builder.In("type", ReviewTypeApprove, ReviewTypeReject))).
```

Comment:
I'm unsure we can update Dismissed directly since @pat-s related PR.

Reply:
what is related PR number?

Reply:
@lunny cant find it https://github.com/go-gitea/gitea/pulls?q=is%3Apr+author%3Apat-s

Reply:
Sorry, I made a mistake. It's #28498 from @jpraet.

Reply:
@lunny the "stale" property is not touched and is handled differently. so it does not make any issues here at all :)